vnext/ReactUWP/Views/ScrollViewManager.cpp

```diff
@@ -28,6 +15,268 @@ enum class ScrollViewCommands
+        scrollViewer.VerticalScrollMode(horizontal ? winrt::ScrollMode::Disabled : winrt::ScrollMode::Auto);
+        ScrollViewUWPImplementation(scrollViewer).SetHorizontal(horizontal);
+      }
+    }
+    else if (propertyName == "showsHorizontalScrollIndicator")
+    {
+      const auto [valid, showsHorizontalScrollIndicator] = getPropertyAndValidity(propertyValue, true);
+      if (valid)
+      {
+        scrollViewer.HorizontalScrollBarVisibility(showsHorizontalScrollIndicator ? winrt::ScrollBarVisibility::Visible : winrt::ScrollBarVisibility::Hidden);
+      }
+    }
+    else if (propertyName == "showsVerticalScrollIndicator")
+    {
+      const auto [valid, showsVerticalScrollIndicator] = getPropertyAndValidity(propertyValue, true);
+      if (valid)
+      {
+        scrollViewer.VerticalScrollBarVisibility(showsVerticalScrollIndicator ? winrt::ScrollBarVisibility::Visible : winrt::ScrollBarVisibility::Hidden);
+      }
+    }
+    else if (propertyName == "minimumZoomScale")
+    {
+      const auto [valid, minimumZoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.MinZoomFactor(static_cast<float>(minimumZoomScale));
+      }
+    }
+    else if (propertyName == "maximumZoomScale")
+    {
+      const auto [valid, maximumZoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.MaxZoomFactor(static_cast<float>(maximumZoomScale));
+      }
+    }
+    else if (propertyName == "zoomScale")
+    {
+      const auto [valid, zoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.ChangeView(nullptr, nullptr, static_cast<float>(zoomScale));
+      }
+    }
+    else if (propertyName == "snapToInterval")
+    {
+      const auto [valid, snapToInterval] = getPropertyAndValidity(propertyValue, 0.0);
+      if (valid)
+      {
+        ScrollViewUWPImplementation(scrollViewer).SnapToInterval(static_cast<float>(snapToInterval));
+      }
+    }
+    else if (propertyName == "snapToOffsets")
+    {
+      const auto [valid, snapToOffsets] = getPropertyAndValidity(propertyValue, winrt::single_threaded_vector<float>());
+      if (valid)
+      {
+        ScrollViewUWPImplementation(scrollViewer).SnapToOffsets(snapToOffsets.GetView());
+      }
+    }
+    else if (propertyName == "snapToAlignment")
+    {
+      const auto [valid, snapToAlignment] = getPropertyAndValidity(propertyValue, winrt::SnapPointsAlignment::Near);
+      if (valid)
+      {
+        scrollViewer.HorizontalSnapPointsAlignment(snapToAlignment);
+        scrollViewer.VerticalSnapPointsAlignment(snapToAlignment);
+      }
+    }
+  }
+
+  Super::updateProperties(std::move(reactDiffMap));
+  m_updating = false;
+}
+
+template <typename T>
+std::tuple<bool, T> ScrollViewShadowNode::getPropertyAndValidity(folly::dynamic propertyValue, T defaultValue)
```

Comment:
>getPropertyAndValidity [](start = 42, length = 22)

there is something like this we probably want to future add that will convert and set values, taking in the dependency prop and dynamic (and maybe more args to support calling xamldirect).  This is probably fine but seems pretty generic that we could use more broadly except for the last couple things in here

Reply:
I agree we should make a broader helper, but I'm not positive what it would look like. I can see about factoring this out.